### PR TITLE
Add nrows parameter to pandas.read_excel()

### LINKED ISSUE #16645: Pandas read_excel: only read first few lines
#### Code Sample, a copy-pastable example if possible

```python
workbook_dataframe = pd.read_excel(workbook_filename, nrows = 10)
```
#### Problem description

Using pandas read_excel on about 100 excel files - some are large - I want to read the first few lines of each (header and first few rows of data).

The above doesn't work but illustrates the goal (example reading 10 data rows).

### PR DESCRIPTION
**Pull request by alysivji**
- [x] closes #16645 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

I was working on this a few months back, but got busy. Apologies. I'm free until January so thought I'd pick this back up.

### Changes
* Added code for `nrows` parameter (with tests)
* Changed order of parameters to match `pandas.read_csv()` signature (across entire `pandas/io/excel.py` file)
* Increased readability by having one parameter per line
* Updated docstring to match new order (+ new entry for `nrows` param)

I think that covered all the outstanding issues from the [previous (closed) PR](https://github.com/pandas-dev/pandas/pull/16672). Please let me know if you need anything changed or modified.

### CODE REVIEW

**Comment by foresmac** [comment]
LGTM; Travis build issues appear to be related to quote issues and not test failures.

**Review by foresmac — COMMENTED** [review]
LGTM

**Comment by alysivji** [comment]
Hi @jreback @gfyoung 
Could you please take a look at this when you have some time?

Thanks!

**Comment by gfyoung** [comment]
@alysivji : Thanks for the ping.  A couple of points:

1) We do need the Travis builds to pass before we can merge this.  I'll double-check to see if you need to do anything to fix that.

2) Your PR is for adding `nrows` to `read_excel`, but you also changed up the signature.  IMO that's a little too much API changing in one PR for us to review.  Let's break up this PR into two by separating out your `read_excel` signature changes into a separate PR.

I'm totally onboard for adding `nrows` but need to review your signature changes separately.

File: pandas/io/excel.py [hunk 1]
```diff
@@ -78,24 +78,28 @@
...
-names : array-like, default None
-    List of column names to use. If file contains no header row,
-    then you should explicitly pass header=None
-converters : dict, default None
-    Dict of functions for converting values in certain columns. Keys can
-    either be integers or column labels, values are functions that take one
-    input argument, the Excel cell content, and return the transformed
-    content.
+parse_cols : int or list, default None
+    .. deprecated:: 0.21.0
```
Comment (jreback) [inline pandas/io/excel.py]: I think you need a blank line before the deprecated

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
I am ok with the re-aranging of parameters in this PR. pls rebase.

File: pandas/io/excel.py (hunk 1 above)
Comment (alysivji) [reply pandas/io/excel.py]: This has been fixed, also added a blank line in the `sheetname` section. Thanks.

**Comment by alysivji** [comment]
@jreback How does this look? Any changes needed?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Comment by Sabr3tooth** [comment]
Hi folks,

Reading the Whats New for Pandas 0.22.0 as well as checking the pandas.read_excel page I note that nrows param is not yet available for read_excel(..).  Is there any indication as to when as well as what version of Pandas it will be added?  Thanks in advance to the people involved in adding this functionality, keep up the great work!

**Comment by alysivji** [comment]
@Sabr3tooth Pandas 0.22 had a [single API breaking change](https://pandas.pydata.org/pandas-docs/stable/whatsnew.html#v0-22-0-december-29-2017).

This enhancement will be in 0.23 based on the [whatsnew in master](https://github.com/pandas-dev/pandas/blob/master/doc/source/whatsnew/v0.23.0.txt)

**Comment by Sabr3tooth** [comment]
Thanks, looking forward to seeing it in Pandas!

**Pull request merged**